nailgun/entities.py

```diff
@@ -267,41 +267,39 @@ def path(self, which=None):
             )
         return super(ActivationKey, self).path(which)
 
-    def add_subscriptions(self, payload):
+    def add_subscriptions(self, synchronous=True, **kwargs):
         """Helper for adding subscriptions to activation key.
 
-        :param payload: Parameters that are encoded to JSON and passed in
-            with the request. See the API documentation page for a list of
-            parameters and their descriptions.
+        :param synchronous: What should happen if the server returns an HTTP
+            202 (accepted) status code? Wait for the task to complete if
+            ``True``. Immediately return the server's response otherwise.
+        :param kwargs: Arguments to pass to requests.
         :returns: The server's response, with all JSON decoded.
         :raises: ``requests.exceptions.HTTPError`` If the server responds with
             an HTTP 4XX or 5XX message.
 
         """
-        response = client.put(
-            self.path('add_subscriptions'),
-            payload,
-            **self._server_config.get_client_kwargs()
-        )
-        return _handle_response(response, self._server_config)
+        kwargs = kwargs.copy()  # shadow the passed-in kwargs
```

Comment:
Let's say the user wants to call `AbstractDockerContainer.logs` (or
`DockerHubContainer.logs`) on several containers. The client could put together
a request that looks like this:

```python
lost_list = [
    container.logs(data={'tail': 50, 'stdout': True, 'stderr': False})
    for container in containers
]
```

This works just fine. Another way to write the above is this:

```python
data = {'tail': 50, 'stdout': True, 'stderr': False}
lost_list = [container.logs(data=data) for container in containers]
```

In this latter form, we depend upon `data` being the same for every single
request. Right? It would be pretty weird if we passed `data` in to a method and
the method altered `data`. In fact, if the method altered `data`, that would
break the latter way of writing our code!

`kwargs.copy()` prevents `data` from being altered. Check this out:

```python
>>> def logs(data):
...     data.update({'tail': 1000})
... 
>>> data = {'tail': 50, 'stdout': True, 'stderr': False}
>>> logs(data)
>>> data
{'tail': 1000, 'stderr': False, 'stdout': True}
```

Uh oh! Using `kwargs.copy()` avoid this issue:

```python
>>> def logs(data):
...     data = data.copy()  # shadow the passed-in argument
...     data.update({'tail': 1000})
... 
>>> data = {'tail': 50, 'stdout': True, 'stderr': False}
>>> logs(data)
>>> data
{'tail': 50, 'stderr': False, 'stdout': True}
```